### ISSUE TITLE
Following links between metabolite pages requires a manual refresh

### ISSUE THREAD

**Issue by JonathanRob** [MetabolicAtlas/MetabolicAtlas#646]
When I am viewing a page for a metabolite in the GEM Browser and select a link to another metabolite, it looks like I am taken to a new page, but it ends up being the same page that appears. Only after a manual refresh does the page corresponding to the selected metabolite appear.

The same behavior doesn't seem to occur for reactions or genes, so it seems to be specific to the metabolite pages.

This also has the effect of disabling the function of the "See reactions from all compartments" button on a metabolite page. Or maybe that's a separate issue.

**Comment by mihai-sysbio** [comment]
Thanks for reporting this @JonathanRob.

**Comment by e0** [comment]
Looks like this is no longer a bug. I think it was resolved with @inghylt's [PR](https://github.com/MetabolicAtlas/MetabolicAtlas/pull/632) for refactoring the gem browser components using a common layout.

**Comment by e0** [comment]
In retrospect, since no further work was needed, the adjusted estimate is 0.

**Comment by JonathanRob** [comment]
I think this issue needs to be re-opened, I'm still seeing some effects like those described.

For example, if I am on the GEM explorer page for [MAR05029](https://metabolicatlas.org/explore/Human-GEM/gem-browser/reaction/MAR05029) and click the link to MAR07675 in the `Related reaction(s)` field, it updates most of the page but the `Related reactions(s)` still includes MAR07675. When I manually refresh the page, it is fixed.

Also, the `See reactions from all compartments` button on metabolite GEM browser pages still doesn't seem to work - it only shows reactions from the current compartment.

**Comment by mihai-sysbio** [comment]
> I think this issue needs to be re-opened, I'm still seeing some effects like those described.
> 
> For example, if I am on the GEM explorer page for [MAR05029](https://metabolicatlas.org/explore/Human-GEM/gem-browser/reaction/MAR05029) and click the link to MAR07675 in the `Related reaction(s)` field, it updates most of the page but the `Related reactions(s)` still includes MAR07675. When I manually refresh the page, it is fixed.
> 
> Also, the `See reactions from all compartments` button on metabolite GEM browser pages still doesn't seem to work - it only shows reactions from the current compartment.

Thanks for the detailed walk-through. With a little bit of an insight into how the system works, i.e. the related reactions are fetched via a separate query, I think both bugs should be reported as distinct issues.
